app/assets/javascripts/services.js

```diff
@@ -1,5 +1,8 @@
 angular.module("barter").factory("UserService", function($resource){
-   return $resource("/users/:id");
+   return $resource("/users/:id", {id:'@id'},
```

Comment:
Figured this was worth a line note - the {id: @id} bit says use the id field in the object to construct the URL. So if I were to call UserService.update(user) where user is some object like {id:11, name:'steven',....} then it'll send a put request to the url /users/11, which is what the rails side expects. You will probably need to change the "strong params" on the rails side not to expect the params wrapped in "user" - i.e. remove the require(:user) from the method and just have the permit part. This is because your angular controller is sending back the object as json like 
{id: 1, username: "name", email: "name@example.com", location:.....} - it's not wrapped in "user"